data/content.go

```diff
@@ -1315,16 +1413,29 @@ var ContentHeuristics = map[string]*Heuristics{
 	".plist": &Heuristics{
 		rule.Or(
 			rule.MatchingLanguages("XML Property List"),
-			regex.MustCompileMultiline(`<!DOCTYPE\s+plist`),
+			regex.MustCompileRuby(`^\s*(?:<\?xml\s|<!DOCTYPE\s+plist|<plist(?:\s+version\s*=\s*(["'])\d+(?:\.\d+)?\1)?\s*>\s*$)`),
```

Comment:
The problems discussed in [this comment ](https://github.com/go-enry/go-enry/pull/169#issuecomment-1708840755) are most likely coming from the fact that the new heuristic rule for XML Property List now contains a backreference (`\1`).
It's there to make sure the pattern matches the same type of quotation marks (single or double).

Seems like replacing `\1` with `["']` would be a simple way to handle this. I'm not an expert in the syntaxes at hand, but I don't see how using any quotation marks could create any false positives.

Example in Rubular ([link](https://rubular.com/r/PWAYanXCAp4ABp)) using `ff-man.plist` introduced [here](https://github.com/jorendorff/linguist/blob/0f17801fa3032b32d24333ceac86846deb913a37/samples/XML%20Property%20List/ff-man.plist](https://github.com/github-linguist/linguist/commit/0f17801fa3032b32d24333ceac86846deb913a37)):
![image](https://github.com/go-enry/go-enry/assets/31558169/c2e95866-47e0-41ee-b99c-80191a055d07)

Reply:
Thank you for sharing these detailed - it's very useful to know that more Linguist upstream features rely on regex syntax that is definitely not going to be supported by the current default [Go regex engine](https://github.com/google/re2/wiki/Syntax).

I don't think that re-writing cases like this on our side is a sustainable policy: so far as they were few, we just skipped such rules at runtime and [documented the divergence](https://github.com/go-enry/go-enry/#divergences-from-linguist) for the default configuration (RE2). The clients who need the precision would build their applications with oniguruma engine (`-tag oniguruma`, see https://github.com/go-enry/go-enry/#faster-regexp-engine-optional ) that is faster and does support this feature but comes at a price of going though the native bindings. 

As for the rule syntax change - if it does not introduce any changes in the results, not using backreferences is faster and thus it should be proposed as a PR to the Linguist itself.

It means that for the next release we'll probably have to go "the usual route" (skip&document) this case, as soon as oniguruma CI profile passes (after this PR is rebased on #170, that ATM is stuck as I was planning to look into Python CI next week - any help will be greatly appreciated).

This makes me think that in the future versions of enry it'll be important to have an option to use a feature-rich pure-Go regex engine! One option is adding https://github.com/dlclark/regexp2, measuring it's runtime performance impact (though our [benchmark](https://github.com/go-enry/go-enry#benchmarks)) and checking how many of the existing divergences that would get rid of to make a decision if it could be used as a default.

Hope this helps!

Reply:
> I don't think that re-writing cases like this on our side is a sustainable policy for enry project: so far as they were few, we just skipped such rules at runtime and [documented the divergence](https://github.com/go-enry/go-enry/#divergences-from-linguist) for the default configuration (RE2).

Sounds good. When you say skipping, do you mean that all files with that extension will skip the regex-heuristic step and go directly to the baysian classifier?

> As for the rule syntax change - if it does not introduce any changes in the results, not using backreferences is faster and thus it should be proposed as a PR to the Linguist itself.

Could you create an issue in the Linguist project itself listing what regex features should be kept to a minimum like backreference and lookarounds and explain why it matters for Linguist, Enry and Github in general? It would be easier to have an issue to refer to when submitting PRs that try to remove those features where applicable.